### DISCUSSION TITLE
ScriptReference/MonoBehaviour.OnWillRenderObject

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#17] [Notes]
# ScriptReference/MonoBehaviour.OnWillRenderObject

https://docs.unity3d.com/ScriptReference/MonoBehaviour.OnWillRenderObject.html

**Comment by vertxxyz** [comment]
This is not called in Edit Mode without the [`ExecuteInEditMode`](https://docs.unity3d.com/ScriptReference/ExecuteInEditMode.html) attribute.

**Comment by BelovK-Dev** [comment]
This doesn't work with empty object. Need to be used with Mesh Renderer component.

**Reply by BobyStar** [reply]
You can also use most if not all "Renderer" components. Sprite Renderer with no Sprite works well if you want an "invisible" empty that gets this call.

**Reply by BelovK-Dev** [reply]
Okay, I don't think I've tried that. I'll keep that in mind. Thanks!
